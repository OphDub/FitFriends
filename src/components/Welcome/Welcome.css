.Welcome {
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  background: linear-gradient(rgba(0, 0, 0, 0.45), rgba(0, 0, 0, 0.45)), url("https://static.pexels.com/photos/685534/pexels-photo-685534.jpeg");
  background-repeat: no-repeat;
  height: 100vh; }

.Welcome-container {
  display: flex;
  flex-direction: column;
  align-items: center;
  background-color: #D8D8D8;
  padding: 15px;
  border-radius: 5px; }

.fitfriends-text {
  color: white;
  text-transform: uppercase;
  font-size: 40px; }

.fitfriends-text-emphasis {
  color: #48E0EA; }

.welcome-signup {
  display: flex;
  flex-direction: column;
  width: 100%; }

.welcome-signup a {
  width: 100%; }

.welcome-signup h5 {
  margin: 10px;
  align-self: center; }

.sign-up-btn {
  border: none;
  padding: 8px;
  text-transform: uppercase;
  border-radius: 5px;
  outline: none;
  width: 100%; }

.sign-up-btn:hover {
  background-color: #48E0EA;
  color: white;
  cursor: pointer; }
