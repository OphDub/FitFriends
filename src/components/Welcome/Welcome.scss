
.Welcome {
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  background: linear-gradient(
    rgba(0, 0, 0, 0.45),
    rgba(0, 0, 0, 0.45)
  ),url('https://static.pexels.com/photos/685534/pexels-photo-685534.jpeg');
  background-repeat: no-repeat;
  height: 100%;
}