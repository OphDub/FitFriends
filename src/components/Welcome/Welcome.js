import React, { Component } from 'react';
import './Welcome.css';
import Control from '../Control/Control';
import { NavLink } from 'react-router-dom';

class Welcome extends Component {
  constructor () {
    super();
  }

  handleReroute = () => {
    return this.props.history.push('/home');
  }

  render () {
    return (
      <div className="Welcome">
          <h2 className="fitfriends-text">
            <span className="fitfriends-text-emphasis">
              Fit
            </span>
              Friends
          </h2>
        <div className="Welcome-container">
          <Control handleReroute={this.handleReroute}/>
          <div className="welcome-signup">
            <h5>New? Make an account:</h5>
            <NavLink className="sign-up-link"
              to="/signup">
              <button className="sign-up-btn">
                Sign Up
              </button>
            </NavLink>
          </div>
        </div>
      </div>
    )
  }
};

export default Welcome

