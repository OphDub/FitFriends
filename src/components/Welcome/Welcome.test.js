/* eslint-disable */
import React from 'react';
import { shallow } from 'enzyme';
import Welcome from './Welcome';

describe('Welcome', () => {
  it('should match snapshot', () => {
    const renderedComponent = shallow(<Welcome />);

    expect(renderedComponent).toMatchSnapshot();
  });
});
