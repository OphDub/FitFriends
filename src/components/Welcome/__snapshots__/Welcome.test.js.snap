// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Welcome should match snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Welcome />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h2
          className="fitfriends-text"
>
          <span
                    className="fitfriends-text-emphasis"
          >
                    Fit
          </span>
          Friends
</h2>,
        <div
          className="Welcome-container"
>
          <withRouter(Connect(Control)) />
          <div
                    className="welcome-signup"
          >
                    <h5>
                              New? Make an account:
                    </h5>
                    <NavLink
                              activeClassName="active"
                              ariaCurrent="true"
                              className="sign-up-link"
                              to="/signup"
                    >
                              <button
                                        className="sign-up-btn"
                              >
                                        Sign Up
                              </button>
                    </NavLink>
          </div>
</div>,
      ],
      "className": "Welcome",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <span
              className="fitfriends-text-emphasis"
>
              Fit
</span>,
            "Friends",
          ],
          "className": "fitfriends-text",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Fit",
              "className": "fitfriends-text-emphasis",
            },
            "ref": null,
            "rendered": "Fit",
            "type": "span",
          },
          "Friends",
        ],
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <withRouter(Connect(Control)) />,
            <div
              className="welcome-signup"
>
              <h5>
                            New? Make an account:
              </h5>
              <NavLink
                            activeClassName="active"
                            ariaCurrent="true"
                            className="sign-up-link"
                            to="/signup"
              >
                            <button
                                          className="sign-up-btn"
                            >
                                          Sign Up
                            </button>
              </NavLink>
</div>,
          ],
          "className": "Welcome-container",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <h5>
                  New? Make an account:
</h5>,
                <NavLink
                  activeClassName="active"
                  ariaCurrent="true"
                  className="sign-up-link"
                  to="/signup"
>
                  <button
                                    className="sign-up-btn"
                  >
                                    Sign Up
                  </button>
</NavLink>,
              ],
              "className": "welcome-signup",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "New? Make an account:",
                },
                "ref": null,
                "rendered": "New? Make an account:",
                "type": "h5",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "activeClassName": "active",
                  "ariaCurrent": "true",
                  "children": <button
                    className="sign-up-btn"
>
                    Sign Up
</button>,
                  "className": "sign-up-link",
                  "to": "/signup",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Sign Up",
                    "className": "sign-up-btn",
                  },
                  "ref": null,
                  "rendered": "Sign Up",
                  "type": "button",
                },
                "type": [Function],
              },
            ],
            "type": "div",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h2
            className="fitfriends-text"
>
            <span
                        className="fitfriends-text-emphasis"
            >
                        Fit
            </span>
            Friends
</h2>,
          <div
            className="Welcome-container"
>
            <withRouter(Connect(Control)) />
            <div
                        className="welcome-signup"
            >
                        <h5>
                                    New? Make an account:
                        </h5>
                        <NavLink
                                    activeClassName="active"
                                    ariaCurrent="true"
                                    className="sign-up-link"
                                    to="/signup"
                        >
                                    <button
                                                className="sign-up-btn"
                                    >
                                                Sign Up
                                    </button>
                        </NavLink>
            </div>
</div>,
        ],
        "className": "Welcome",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <span
                className="fitfriends-text-emphasis"
>
                Fit
</span>,
              "Friends",
            ],
            "className": "fitfriends-text",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Fit",
                "className": "fitfriends-text-emphasis",
              },
              "ref": null,
              "rendered": "Fit",
              "type": "span",
            },
            "Friends",
          ],
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <withRouter(Connect(Control)) />,
              <div
                className="welcome-signup"
>
                <h5>
                                New? Make an account:
                </h5>
                <NavLink
                                activeClassName="active"
                                ariaCurrent="true"
                                className="sign-up-link"
                                to="/signup"
                >
                                <button
                                                className="sign-up-btn"
                                >
                                                Sign Up
                                </button>
                </NavLink>
</div>,
            ],
            "className": "Welcome-container",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <h5>
                    New? Make an account:
</h5>,
                  <NavLink
                    activeClassName="active"
                    ariaCurrent="true"
                    className="sign-up-link"
                    to="/signup"
>
                    <button
                                        className="sign-up-btn"
                    >
                                        Sign Up
                    </button>
</NavLink>,
                ],
                "className": "welcome-signup",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "New? Make an account:",
                  },
                  "ref": null,
                  "rendered": "New? Make an account:",
                  "type": "h5",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "activeClassName": "active",
                    "ariaCurrent": "true",
                    "children": <button
                      className="sign-up-btn"
>
                      Sign Up
</button>,
                    "className": "sign-up-link",
                    "to": "/signup",
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Sign Up",
                      "className": "sign-up-btn",
                    },
                    "ref": null,
                    "rendered": "Sign Up",
                    "type": "button",
                  },
                  "type": [Function],
                },
              ],
              "type": "div",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
